### PR TITLE
Support the request and response union body in rlc

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1926

### CODE REVIEW

File: packages/typespec-test/test/healthinsight/generated/typespec-ts/src/responses.ts [hunk 1]
```diff
@@ -57,4 +57,5 @@ export interface CreateJobDefaultResponse extends HttpResponse {
 /** The final response for long-running createJob operation */
 export interface CreateJobLogicalResponse extends HttpResponse {
   status: "200";
+  body: TrialMatcherResultOutput;
```
Comment (MaryGao) [inline packages/typespec-test/test/healthinsight/generated/typespec-ts/src/responses.ts:60]: Fix the lro logical reponse body missing issue.

File: packages/typespec-ts/test/commands/cadl-ranch-list.ts [hunk 2]
```diff
@@ -51,7 +55,8 @@ export const cadls: CadlRanchConfig[] = [
   },
   {
     outputPath: "dictionary",
-    inputPath: "type/dictionary"
+    inputPath: "type/dictionary",
+    debug: true
```
Comment (MaryGao) [inline packages/typespec-ts/test/commands/cadl-ranch-list.ts:59]:
```suggestion
     inputPath: "type/dictionary"
```

File: packages/typespec-ts/test/commands/cadl-ranch-list.ts [hunk 3]
```diff
@@ -51,7 +55,8 @@ export const cadls: CadlRanchConfig[] = [
   },
   {
     outputPath: "dictionary",
-    inputPath: "type/dictionary"
+    inputPath: "type/dictionary",
```
Comment (MaryGao) [inline packages/typespec-ts/test/commands/cadl-ranch-list.ts:58]:
```suggestion
```

File: packages/typespec-ts/test/commands/cadl-ranch-list.ts [hunk 4]
```diff
@@ -51,7 +55,7 @@ export const cadls: CadlRanchConfig[] = [
   },
   {
     outputPath: "dictionary",
-    inputPath: "type/dictionary"
+     inputPath: "type/dictionary"
```
Comment (MaryGao) [inline packages/typespec-ts/test/commands/cadl-ranch-list.ts:58]:
```suggestion
    inputPath: "type/dictionary"
```

**Review by qiaozha — APPROVED** [review]

**Pull request merged**